### PR TITLE
infra: run checkstyle in ru locale

### PR DESCRIPTION
**Pull request by romani**
related to https://github.com/checkstyle/checkstyle/pull/11280

### CODE REVIEW

File: .ci/validation.sh [hunk 1]
```diff
@@ -167,6 +167,11 @@ test-tr)
     -DargLine='-Duser.language=tr -Duser.country=TR -Xms1024m -Xmx2048m'
   ;;
 
+test-ru)
+  mvn -e --no-transfer-progress clean integration-test failsafe:verify \
+    -DargLine='-Duser.language=ru -Duser.country=RU -Xms1024m -Xmx2048m'
```
Comment (rnveach) [inline .ci/validation.sh:172]: Can we not rework these so instead of 20 "cmd test-XX", can we do "cmd test-language XX YY"?

Comment (romani) [reply .ci/validation.sh:172]: @strkkk , do you know why CI for Run is green?

Comment (strkkk) [reply .ci/validation.sh:172]: No idea

Comment (romani) [reply .ci/validation.sh:172]: We need find way to fail CI in other cases we never resolve this issue

Comment (rnveach) [reply .ci/validation.sh:172]: Running `integration-test failsafe:verify -DargLine='-Duser.language=ru -Duser.country=RU'` on my local,

I am getting output saying my locale is still english.

````
[INFO] Running com.puppycrawl.tools.checkstyle.api.ViolationTest
1: en_US
2: en
[WARNING] Tests run: 22, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0.1 s - in com.puppycrawl.tools.checkstyle.api.ViolationTest
````

````
diff --git a/src/test/java/com/puppycrawl/tools/checkstyle/api/ViolationTest.java b/src/test/java/com/puppycrawl/tools/checkstyle/api/ViolationTest.java
index efccf13..01cdacb 100644
--- a/src/test/java/com/puppycrawl/tools/checkstyle/api/ViolationTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/api/ViolationTest.java
@@ -84,10 +84,13 @@
      */
     @Test
     public void testLocaleIsSupported() {
+        System.out.println("1: " + DEFAULT_LOCALE);
         final String language = DEFAULT_LOCALE.getLanguage();
+        System.out.println("2: " + language);
         assumeFalse(language.isEmpty() || Locale.ENGLISH.getLanguage().equals(language),
                 "Custom locale not set");
         final Violation violation = createSampleViolation();
+        System.out.println("3: " + violation.getViolation());
         assertWithMessage("Unsupported language: %s", DEFAULT_LOCALE)
                 .that(violation.getViolation())
                 .isNotEqualTo("Empty statement.");
````

Comment (romani) [reply .ci/validation.sh:172]: It means that plugin does not use value of argLine, can you change plugin in pom.xml to get locale to be set.

Comment (rnveach) [reply .ci/validation.sh:172]: Russian run in CI says items were skipped.
> [WARNING] Tests run: 22, Failures: 0, Errors: 0, Skipped: 2, Time elapsed: 0.227 s - in com.puppycrawl.tools.checkstyle.api.ViolationTest

3 tests have the assumes in this class, so it doesn't say if we can connect this.

Is there a way to make assumes print out if they are causing the test to skip?

Comment (rnveach) [reply .ci/validation.sh:172]: Change in pom is producing errors.
````
diff --git a/pom.xml b/pom.xml
index 34aee2a..4ec154a 100644
--- a/pom.xml
+++ b/pom.xml
@@ -197,7 +197,7 @@
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
-    <surefire.options />
+    <surefire.options>-Duser.language=ru -Duser.country=RU</surefire.options>
     <projectVersion>${project.version}</projectVersion>
     <antlr4.version>4.9.3</antlr4.version>
     <maven.site.plugin.version>3.11.0</maven.site.plugin.version>
````

````
[INFO] Running com.puppycrawl.tools.checkstyle.api.ViolationTest
1: ru_RU
2: ru
3: Empty statement.
[ERROR] Tests run: 22, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.147 s <<< FAILURE! - in com.puppycrawl.tools.checkstyle.api.ViolationTest
[ERROR] testLocaleIsSupported  Time elapsed: 0.062 s  <<< FAILURE!
com.google.common.truth.AssertionErrorWithFacts: 
Unsupported language: ru_RU
expected not to be: Empty statement.
	at com.puppycrawl.tools.checkstyle.api.ViolationTest.testLocaleIsSupported(ViolationTest.java:96)
````

Comment (rnveach) [reply .ci/validation.sh:172]: Full failure list:
````
[ERROR] Failures: 
[ERROR]   ViolationTest.testLocaleIsSupported:96 Unsupported language: ru_RU
expected not to be: Empty statement.
[ERROR] Errors: 
[ERROR]   DefaultLoggerTest.testLocaleIsSupported:254 » NoSuchMethod com.puppycrawl.tool...


[ERROR] testLocaleIsSupported  Time elapsed: 0 s  <<< ERROR!
java.lang.NoSuchMethodException: com.puppycrawl.tools.checkstyle.DefaultLogger$LocalizedMessage.<init>(java.lang.String, [Ljava.lang.String;)
	at com.puppycrawl.tools.checkstyle.DefaultLoggerTest.testLocaleIsSupported(DefaultLoggerTest.java:254)
````

Comment (rnveach) [reply .ci/validation.sh:172]: Seems strange that `surefire.options` is pulled into `argLine` but setting `argLine` directly doesn't work...

Comment (rnveach) [reply .ci/validation.sh:172]: https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=8630&view=logs&j=3b0b6035-d26e-5411-a529-6e236284dc45&t=fdcfc88d-226d-5705-159e-79095a68acd9
>  (s) argLine = -Dfile.encoding=UTF-8 

Seems like argLine isn't being replaced.

Comment (rnveach) [reply .ci/validation.sh]: CI is now reproducing my local.

**Comment by romani** [comment]
conflict with bdd style of testing.
we can disable inlined message validation for any non english locale, so it will be compared with expected messages array only - users might not catch a problem until push code to CI.
if we enforce EN locale for all, it will resolve all problems for all users, and we need to figureout only how override that in CI and execute CI in special language(with disablement of inline message validation so such locale specific executions). 

removing pain from random contributors looks like a winning concept.

so now it is failing:
```
[ERROR] testDeclarationOrderInterfaceMemberScopeIsPublic  Time elapsed: 0.004 s  <<< FAILURE!
com.google.common.truth.AssertionErrorWithFacts: 
Actual and expected violations differ.
expected to match: 21:(?:\d+:)?\s.*Static variable definition in wrong order.*
but was          : 21:3: La définition des variables statiques n'apparaît pas dans le bon ordre.
	at com.puppycrawl.tools.checkstyle.checks.coding.DeclarationOrderCheckTest.
testDeclarationOrderInterfaceMemberScopeIsPublic(DeclarationOrderCheckTest.java:215)
```

File: .ci/validation.sh (hunk 1 above)
Comment (romani) [reply .ci/validation.sh]: >ViolationTest.testLocaleIsSupported:96 Unsupported language: ru_RU expected not to be: Empty statement.

ok, as we fix it, this should not be a failure.
Lets keep RU (or we can select some other very unlikely used language) execution as prove that tests are passing on it.

Comment (romani) [reply .ci/validation.sh]: WE can take 1st from https://www.oracle.com/java/technologies/javase/java8locales.html


Language (ISO 639) | Country (ISO 3166) | Variant | Script (ISO 15924) | Language Tag | Note
-- | -- | -- | -- | -- | --
Albanian (sq) | Albania (AL) |   | (Latn) | sq-AL |

**Comment by romani** [comment]
@pbludov , @strkkk , @rnveach , @nick-mancuso , @baratali , please vote

**Comment by nick-mancuso** [comment]
> push code to CI.
if we enforce EN locale for all, it will resolve all problems for all users, and we need to figureout only how override that in CI and execute CI in special language(with disablement of inline message validation so such locale specific executions).

I am for this option, we could just have some testing property/ resource file that specifies option that can be overridden as needed.

**Comment by rnveach** [comment]
as long as BDD is in English we will have to force English. We still need to test languages in CI to ensure all messages work.

**Comment by baratali** [comment]
> please vote on what is preferred solution to let us create issue and start doing it.

this option seems better:
> if we enforce EN locale for all, it will resolve all problems for all users, and we need to figureout only how override that in CI and execute CI in special language(with disablement of inline message validation so such locale specific executions).

**Comment by romani** [comment]
@strkkk , @pbludov , ^^ lets finalize agreement on this topic

**Comment by romani** [comment]
@strkkk , please rebase and resolve conflict.

>Maybe we need to build and execute Checksyle with a small input file and verify that the output is different for each supported language

We can make CLI execution to generate output and validate that it same predefined output. So it will be CI only execution, outside of mvn build command/phases.

**Comment by strkkk** [comment]
> @strkkk , please rebase and resolve conflict.

rebase what?

**Comment by rnveach** [comment]
@romani rebased.

**Comment by romani** [comment]
Omg this is actually my PR :). I am lost.

**Comment by romani** [comment]
Ok, now we have arguments working fine, so all except for RU is red, Ru is green because EN is used instead.

I would rather keep commit that makes it work in our history, and made one more commit to enforce EN language in execution with ability to override for CI.

**Comment by rnveach** [comment]
Yes, I only helped with debugging the issue as it seemed interesting, so I added the extra commits to debug/resolve.

**Comment by rnveach** [comment]
> I would rather keep commits that makes it work in our history

I agree, I would find some commit that we can reference this PR (or the referenced issue) as another history of our talks on this issue. All the commits now current are `infra`.

**Comment by rnveach** [comment]
@romani 
> made one more commit to enforce EN language in execution with ability to override for CI.

Is this all that is required for this PR? I am not seeing clear guidance on what is needed to complete this.

**Comment by romani** [comment]
We need to bind all commits to issue, where we explain all details.
We need commit that enforce EN language, so execution in other languages are not required.
We need somehow to run validation(might be CLI) in Albanian language to confirm that EN is used.
Do we need to run some validation in German ? To make sure language selection is working.

**Comment by romani** [comment]
https://github.com/checkstyle/checkstyle/pull/12039#issuecomment-1210768938

This is exactly a problem we try to resolve in this PR. It is very frustrating experience that project after clone is not able to build without problem.

Goal is to let unit test pass in RU(any locale). So any  contributor, just clone repo, make sure maven and java installed, run `mvn test` and build is passing.

Please look at change of this PR, so we finally are able to reproduce this problem.

We need to update our test execution to run locale US(English) because we use Junit in BDD style, so expected violation message is located in Input file of test. So only English is supported.

Do not hesitate to contact me for more details, we appreciate your help, sorry that we overloaded with PR reviews and do not have time to finish this PR.

**Comment by romani** [comment]
@ZuevKirill95, ∆∆

**Comment by romani** [comment]
@ZuevKirill95, thanks a lot.
Please create your PR, copy to your PR my commit and place as second commit on top your commit with your change.

Please try to avoid static block, and place same code to before class method of AbstractPathTestSupport to make sure it happens before anything executes with Inputs files.

As you open your PR, I will close my PR.

**Comment by ZuevKirill95** [comment]
> Please try to avoid static block, and place same code to before class method of AbstractPathTestSupport to make sure it happens before anything executes with Inputs files.

I'm sorry, but I don't understand which exactly method you meant?
Or I need to do new method in **AbstractPathTestSupport**?

**Comment by romani** [comment]
Yes, new method

**Comment by ZuevKirill95** [comment]
Ok, I got it
Thanks

**Comment by romani** [comment]
closed in favor of https://github.com/checkstyle/checkstyle/pull/12053

**Pull request closed without merging**